### DISCUSSION TITLE
Inconsistent character encoding, same code

### DISCUSSION THREAD

**Discussion by pruuk** [evennia/evennia#2614] [Evennia Questions] [answered]
Hi all,

I have a copy of my mud (I started over again!) going on both my local machine and on a Digital Ocean Droplet. Both are running the same code, but I'm getting inconsistent alignment of my overhead map. My code is using EvTable to split my room description and overhead map into two columns that look like this on my local version:
![image](https://user-images.githubusercontent.com/72707866/148652690-3e6b57ee-a33e-4d09-9e1e-fba456330380.png)
 and like this on the droplet server:
![image](https://user-images.githubusercontent.com/72707866/148652734-7509e5c2-e6e3-46b4-ba51-2e16e13a22fd.png)

I can, of course, change the font to make the output look correct, but it is somewhat annoying that I'm getting misalignment of the characters by default.

Any thoughts? Am I missing a package or something on my Droplet machine that is causing this?

**Comment by pruuk** [comment]
One other thing to note: I logged in using Mudlet to the Droplet server and did not see the same misalignment.

**Reply by Griatch** [reply]
Cool ascii map! The misaligned version has a curious bend to it that is so even that it almost looks intentional :D 

Check the javascript console on your droplet for errors. Maybe it fails to download some font that you can get just fine locally. Maybe @friarzen could chip in with some other suggestions?

**Reply by pruuk** [reply]
I did as you suggested. The difference I see when Im looking at the droplet version running at http://165.227.184.28:4001/webclient/

Is it shows this error:
<img width="428" alt="image" src="https://user-images.githubusercontent.com/72707866/148706559-7c108a84-d3a2-4a61-9c01-99601e8bb976.png">

Clicking on the css:1 link returns a page with the following:

/* latin */
@font-face {
  font-family: 'Consolas';
  font-style: normal;
  font-weight: 400;
  font-display: swap;
  src: url(https://fonts.gstatic.com/l/font?kit=X7nm4bA-A_-9jbjWaza9xMk&skey=3d1eb1871fcc58a1&v=v11) format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}

Funny enough, Consolas is the font I usually swap to (it fixes the alignment issue). The other font that is aligned correctly is Monospace. All the other fonts show varying degrees of weirdness. If it is set to any given font, refreshing the page makes the alignment for that font bad again, and you have to swap to the other one that works to fix the alignment.

**Comment by friarzen** [comment]
My first guess is this is another expression of the errors with the current regex-based approach to crafting the HTML conversion, there are certain cases where multiple inline spaces are broken, since the browser "ignores" multiple spaces and leading spaces unless they are properly converted to &amp;nbsp;  I believe this issue is what first prompted me to look into client-side text rendering.

**Reply by Griatch** [reply]
> My first guess is this is another expression of the errors with the current regex-based approach to crafting the HTML conversion, there are certain cases where multiple inline spaces are broken, since the browser "ignores" multiple spaces and leading spaces unless they are properly converted to &nbsp; I believe this issue is what first prompted me to look into client-side text rendering.

Hm, why would the regex parsing result be expressed differently on a Digital Ocean droplet compared to when running it locally? :thinking:

**Comment by pruuk** [comment, marked as answer]
Well, I gave up on figuring this out. It's probably beyond me for now. The good new is I found a workaround that seems to be functional, and cleaned up my overhead map a bit as well.
